qiskit/aqua/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -314,8 +413,38 @@ def _try_target_num_iterations():
         return result
 
 
+def _oracle_component_to_circuit(oracle: Oracle):
+    """Convert an Oracle to a QuantumCircuit."""
+    circuit = QuantumCircuit(oracle.circuit.num_qubits)
+
+    if isinstance(oracle, TruthTableOracle):
```

Comment:
Oracle was designed as a generic interface - is this specific test on some specific sub-class i.e. TruthTable one, because under this redesign somehow the common info provided via Oracle interface no longer suffices?

Reply:
Somehow, I thought that `TruthTableOracle` does not have `output_register` property, and that's why I added this `if` sentence. However, `TruthTableOracle` does have `output_register` property, and it's my mistake. I will remove it. Thanks!

Reply:
I remembered there was some bug in `QuantumCircuit.extend`, so I had to add the special `if` for `TruthTableOracle`. Now, Julien fixed the bug, so we don't need the special `if` anymore. 
https://github.com/Qiskit/qiskit-terra/pull/5039